.github/workflows/push-chart-releases.yaml

```diff
@@ -0,0 +1,95 @@
+name: Chart Release Push
+
+on:
+  # run after the image build completes
+  workflow_run:
+    workflows:
+      - Image Release Build
+    types:
+      - completed
+  # allow manually triggering it as well, for existing tags
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Git tag to release - with the v prefix"
+        required: true
+
+permissions:
+  # check out source code
+  contents: read
+  # fetch artifact digests
+  actions: read
+  # sign helm chart
+  id-token: write
+
+jobs:
+  push-chart:
+    name: Push Chart
+    runs-on: ubuntu-24.04
+    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch' }}
+    environment: release
+    steps:
+      - name: Get tag
+        id: tag
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            # TODO: validate tag format
+            echo tag="${{ inputs.tag }}" >> $GITHUB_OUTPUT
+          elif [[ "${{ github.event_name }}" == "workflow_run" ]]; then
+            echo tag="${{ github.event.workflow_run.head_branch }}" >> $GITHUB_OUTPUT
+          else
+            echo "Invalid event type"
+            exit 1
+          fi
+
+      - name: Checkout Source Code
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+        with:
+          ref: ${{ steps.tag.outputs.tag }}
+          persist-credentials: false
+
+      - name: Set Environment Variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Download artifact digests
+        uses: actions/download-artifact@v4
+        with:
+          name: Makefile.digests-${{ steps.tag.outputs.tag }}
+          run-id: ${{ github.event.workflow_run.id }}
```

Comment:
Will this work for manual workflow_dispatch? Seems like we might need additional input?

Reply:
Oh right, this wouldn't work indeed. We could add the run ID as an input indeed, but the artifacts expire after 10 days if I'm not mistaken, so this would cause issues anyway.
I'll think about an alternative; could be a simple step that resolves the digest for the given tag (by calling the registry).

Reply:
I'm now using `crane` to get the digests of each image. Let me know if that looks OK to you.